src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionViewModel.cs

```diff
@@ -248,42 +251,47 @@ internal sealed class ConnectionViewModel : BindableBase, IConnectionViewModel {
             return path.ToLower();
         }
 
-        internal static string GetCompletePath(string path) {
+        internal static string GetCompletePath(string path, ICoreShell shell) {
+            // We ALWAYS use HTTPS so no reason to accept anything else.
+            // Default RTVS port is 5444.
             // https://foo:5444 -> https://foo:5444 (no change)
-            // https://foo -> https://foo:443
-            // http://foo -> http://foo:80
-            // http://FOO -> http://foo:80
-            // http://FOO:80 -> http://foo:80
+            // https://foo -> https://foo:5444
+            // http://foo -> https://foo:5444
+            // http://FOO -> https://foo:5444
+            // http://FOO:80 -> https://foo:80
             // foo->https://foo:5444
             Uri uri = null;
             try {
                 Uri.TryCreate(path, UriKind.Absolute, out uri);
             } catch (InvalidOperationException) { } catch (ArgumentException) { } catch (UriFormatException) { }
 
-            if (uri == null || !(uri.IsFile || string.IsNullOrEmpty(uri.Host))) {
-                var hasScheme = uri != null && !string.IsNullOrEmpty(uri.Scheme);
-                var hasPort = uri != null && uri.Port >= 0;
-                var hasPathOrQuery = uri != null && !string.IsNullOrEmpty(uri.PathAndQuery) && uri.PathAndQuery != "/";
+            if(uri != null && uri.IsFile) {
+                return path;
+            }
 
-                if (hasScheme) {
-                    var components = UriComponents.Scheme | UriComponents.UserInfo | UriComponents.Host | UriComponents.Fragment;
+            var userProvidedPath = path;
 
-                    if (hasPathOrQuery) {
-                        components |= UriComponents.Path | UriComponents.Query;
-                    }
+            if (path.IndexOfOrdinal("://") < 0) {
+                path = Invariant($"{Uri.UriSchemeHttps}{Uri.SchemeDelimiter}{path.ToLower()}");
+                try {
+                    Uri.TryCreate(path, UriKind.Absolute, out uri);
+                } catch (InvalidOperationException) { } catch (ArgumentException) { } catch (UriFormatException) { }
+            }
 
-                    if (hasPort) {
-                        components |= UriComponents.StrongPort;
-                    }
+            if (uri != null && !string.IsNullOrEmpty(uri.Host)) {
+                var hasPort = uri.Port >= 0 && (!uri.IsDefaultPort || path.IndexOfOrdinal(":443") > 0);
+                var port = hasPort ? uri.Port : DefaultPort;
+                var hasPathOrQuery = !string.IsNullOrEmpty(uri.PathAndQuery) && uri.PathAndQuery != "/";
+                var mainPart = Invariant($"{Uri.UriSchemeHttps}{Uri.SchemeDelimiter}{uri.Host.ToLower()}:{port}");
```

Comment:
We can `#ifdef` those tests out for Debug then? 

Regarding `UriBuilder`, what's wrong with adding `/` - why does it need to be trimmed? The URLs `http://foo` and `http://foo/` should be identical in all respects. In fact, the latter is a canonical form, and as soon as you put that string into an `Uri` instance, it will also add `/` - e.g. try `new Uri("http://foo").ToString()`.